### ISSUE TITLE
[rustic] Error processing message (wrong-type-argument hash-table-p (:line 0 :character 0))

### ISSUE THREAD

**Issue by the-drunk-coder** [emacs-lsp/lsp-mode#4066]
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [X] You may also try reproduce the issue using clean environment using the following command: `M-x lsp-start-plain`

### Bug description

When using lsp and rustic-mode, the error message in the title appears on any edit to a rust file.

### Steps to reproduce

* import a rust project
* edit an *.rs file

### Expected behavior

No error message

### Which Language Server did you use?

rust-analyzer

### OS

Linux

### Error callstack

```emacs-lisp
Debugger entered--Lisp error: (wrong-type-argument hash-table-p (:line 0 :character 0))
  gethash("line" (:line 0 :character 0))
  lsp--position-compare((:line 0 :character 0) #<hash-table equal 2/2 0x15851f72f953>)
  lsp-point-in-range\?(#<hash-table equal 2/2 0x15851f72f953> #<hash-table equal 2/65 0x15851f730ced>)
  lsp--lens-backend-not-loaded\?(#<hash-table equal 2/65 0x15851f730ced> #<hash-table equal 3/3 0x15851f72f8c1>)
  apply(lsp--lens-backend-not-loaded\? (#<hash-table equal 2/65 0x15851f730ced> #<hash-table equal 3/3 0x15851f72f8c1>))
  #f(compiled-function (&rest args2) #<bytecode -0x11396a67fddce57f>)(#<hash-table equal 3/3 0x15851f72f8c1>)
  #f(compiled-function (elt) #<bytecode 0x1ba478eaed7ae666>)(#<hash-table equal 3/3 0x15851f72f8c1>)
  mapcar(#f(compiled-function (elt) #<bytecode 0x1ba478eaed7ae666>) (#<hash-table equal 3/3 0x15851f72f8c1>))
  #f(compiled-function #'sequence #<bytecode 0x1843e8a80c243634>)(#f(compiled-function (elt) #<bytecode 0x1ba478eaed7ae666>) (#<hash-table equal 3/3 0x15851f72f8c1>))
  apply(#f(compiled-function #'sequence #<bytecode 0x1843e8a80c243634>) #f(compiled-function (elt) #<bytecode 0x1ba478eaed7ae666>) (#<hash-table equal 3/3 0x15851f72f8c1>) nil)
  seq-map(#f(compiled-function (elt) #<bytecode 0x1ba478eaed7ae666>) (#<hash-table equal 3/3 0x15851f72f8c1>))
  seq-filter(#f(compiled-function (&rest args2) #<bytecode -0x11396a67fddce57f>) (#<hash-table equal 3/3 0x15851f72f8c1>))
  lsp-lens--backend-fetch-missing((#<hash-table equal 3/3 0x15851f72f8c1>) #f(compiled-function (lenses version) #<bytecode -0xadea031d47cd816>) 9)
  #f(compiled-function (lenses) #<bytecode -0x14a8680f11071f15>)(((#s(lsp--workspace :ewoc #s(ewoc :buffer #<buffer *lsp-log: rust-analyzer:48051*> :pretty-printer lsp--log-entry-pp :header [[... #5 DL-LIST #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] [#5 ... ... #<marker at 1 in *lsp-log: rust-analyzer:48051*>] "" #<marker at 1 in *lsp-log: rust-analyzer:48051*>] :footer [[... #5 ... #<marker at 27436 in *lsp-log: rust-analyzer:48051*>] [#5 ... DL-LIST #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] "" #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] :dll [[... #5 "" #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] [#5 ... "" #<marker at 1 in *lsp-log: rust-analyzer:48051*>] DL-LIST #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] :last-node nil :hf-pp insert) :server-capabilities #<hash-table equal 26/26 0x15851f554a9b> :registered-server-capabilities (#s(lsp--registered-capability :id "workspace/didChangeWatchedFiles" :method "workspace/didChangeWatchedFiles" :options #<hash-table equal 1/1 0x15851f811829>) #s(lsp--registered-capability :id "workspace/didChangeWatchedFiles" :method "workspace/didChangeWatchedFiles" :options #<hash-table equal 1/1 0x15851f6d113b>)) :root "/home/nik/repositories/megra.rs" :client #s(lsp--client :language-id nil :add-on? nil :new-connection (:connect #f(compiled-function (filter sentinel name environment-fn workspace) #<bytecode 0x111e6683dd226645>) :test\? #f(compiled-function () #<bytecode -0x1dd7b1929c6041bb>)) :ignore-regexps nil :ignore-messages nil :notification-handlers #<hash-table equal 1/65 0x15851f4c3521> :request-handlers #<hash-table equal 0/65 0x15851f4c3adb> :response-handlers #<hash-table eql 0/65 0x15851f4c3dc7> :prefix-function nil :uri-handlers #<hash-table equal 0/65 0x15851f4c40d5> :action-handlers #<hash-table equal 4/65 0x15851f4c3805> :major-modes nil :activation-fn #f(compiled-function (file-name mode) #<bytecode -0x6bb9f9640c09281>) :priority 1 :server-id rust-analyzer :multi-root nil :initialization-options lsp-rust-analyzer--make-init-options :semantic-tokens-faces-overrides (:discard-default-modifiers t :modifiers (... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ...)) :custom-capabilities ((experimental ...)) :library-folders-fn #f(compiled-function (workspace) #<bytecode -0x47e8177e0b048f8>) :before-file-open-fn nil :initialized-fn nil :remote? nil :completion-in-comments? nil :path->uri-fn nil :uri->path-fn nil :environment-fn nil :after-open-fn nil :async-request-handlers #<hash-table equal 0/65 0x15851f4c43ab> :download-server-fn #f(compiled-function (client callback error-callback update\?) #<bytecode 0x1730d1c2ae8ef963>) :download-in-progress? nil :buffers nil :synchronize-sections nil) :host-root nil :proc #<process rust-analyzer> :cmd-proc #<process rust-analyzer> :buffers (#<buffer osc_client.rs>) :semantic-tokens-faces nil :semantic-tokens-modifier-faces nil :extra-client-capabilities nil :status initialized :metadata #<hash-table equal 0/65 0x15851f52ae5f> :watches #<hash-table equal 0/65 0x15851f53ef47> :workspace-folders nil :last-id 0 :status-string nil :shutdown-action nil :diagnostics #<hash-table equal 6/65 0x15851f4ec38d> :work-done-tokens #<hash-table equal 0/65 0x15851f521f8f>) . [#<hash-table equal 3/3 0x15851f72f8c1>])))
  apply(#f(compiled-function (lenses) #<bytecode -0x14a8680f11071f15>) ((#s(lsp--workspace :ewoc #s(ewoc :buffer #<buffer *lsp-log: rust-analyzer:48051*> :pretty-printer lsp--log-entry-pp :header [[... #6 DL-LIST #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] [#6 ... ... #<marker at 1 in *lsp-log: rust-analyzer:48051*>] "" #<marker at 1 in *lsp-log: rust-analyzer:48051*>] :footer [[... #6 ... #<marker at 27436 in *lsp-log: rust-analyzer:48051*>] [#6 ... DL-LIST #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] "" #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] :dll [[... #6 "" #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] [#6 ... "" #<marker at 1 in *lsp-log: rust-analyzer:48051*>] DL-LIST #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] :last-node nil :hf-pp insert) :server-capabilities #<hash-table equal 26/26 0x15851f554a9b> :registered-server-capabilities (#s(lsp--registered-capability :id "workspace/didChangeWatchedFiles" :method "workspace/didChangeWatchedFiles" :options #<hash-table equal 1/1 0x15851f811829>) #s(lsp--registered-capability :id "workspace/didChangeWatchedFiles" :method "workspace/didChangeWatchedFiles" :options #<hash-table equal 1/1 0x15851f6d113b>)) :root "/home/nik/repositories/megra.rs" :client #s(lsp--client :language-id nil :add-on? nil :new-connection (:connect #f(compiled-function (filter sentinel name environment-fn workspace) #<bytecode 0x111e6683dd226645>) :test\? #f(compiled-function () #<bytecode -0x1dd7b1929c6041bb>)) :ignore-regexps nil :ignore-messages nil :notification-handlers #<hash-table equal 1/65 0x15851f4c3521> :request-handlers #<hash-table equal 0/65 0x15851f4c3adb> :response-handlers #<hash-table eql 0/65 0x15851f4c3dc7> :prefix-function nil :uri-handlers #<hash-table equal 0/65 0x15851f4c40d5> :action-handlers #<hash-table equal 4/65 0x15851f4c3805> :major-modes nil :activation-fn #f(compiled-function (file-name mode) #<bytecode -0x6bb9f9640c09281>) :priority 1 :server-id rust-analyzer :multi-root nil :initialization-options lsp-rust-analyzer--make-init-options :semantic-tokens-faces-overrides (:discard-default-modifiers t :modifiers (... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ...)) :custom-capabilities ((experimental ...)) :library-folders-fn #f(compiled-function (workspace) #<bytecode -0x47e8177e0b048f8>) :before-file-open-fn nil :initialized-fn nil :remote? nil :completion-in-comments? nil :path->uri-fn nil :uri->path-fn nil :environment-fn nil :after-open-fn nil :async-request-handlers #<hash-table equal 0/65 0x15851f4c43ab> :download-server-fn #f(compiled-function (client callback error-callback update\?) #<bytecode 0x1730d1c2ae8ef963>) :download-in-progress? nil :buffers nil :synchronize-sections nil) :host-root nil :proc #<process rust-analyzer> :cmd-proc #<process rust-analyzer> :buffers (#<buffer osc_client.rs>) :semantic-tokens-faces nil :semantic-tokens-modifier-faces nil :extra-client-capabilities nil :status initialized :metadata #<hash-table equal 0/65 0x15851f52ae5f> :watches #<hash-table equal 0/65 0x15851f53ef47> :workspace-folders nil :last-id 0 :status-string nil :shutdown-action nil :diagnostics #<hash-table equal 6/65 0x15851f4ec38d> :work-done-tokens #<hash-table equal 0/65 0x15851f521f8f>) . [#<hash-table equal 3/3 0x15851f72f8c1>])))
  #f(compiled-function (&rest args) #<bytecode -0x139b588eb3d731fa>)(((#s(lsp--workspace :ewoc #s(ewoc :buffer #<buffer *lsp-log: rust-analyzer:48051*> :pretty-printer lsp--log-entry-pp :header [[... #5 DL-LIST #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] [#5 ... ... #<marker at 1 in *lsp-log: rust-analyzer:48051*>] "" #<marker at 1 in *lsp-log: rust-analyzer:48051*>] :footer [[... #5 ... #<marker at 27436 in *lsp-log: rust-analyzer:48051*>] [#5 ... DL-LIST #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] "" #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] :dll [[... #5 "" #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] [#5 ... "" #<marker at 1 in *lsp-log: rust-analyzer:48051*>] DL-LIST #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] :last-node nil :hf-pp insert) :server-capabilities #<hash-table equal 26/26 0x15851f554a9b> :registered-server-capabilities (#s(lsp--registered-capability :id "workspace/didChangeWatchedFiles" :method "workspace/didChangeWatchedFiles" :options #<hash-table equal 1/1 0x15851f811829>) #s(lsp--registered-capability :id "workspace/didChangeWatchedFiles" :method "workspace/didChangeWatchedFiles" :options #<hash-table equal 1/1 0x15851f6d113b>)) :root "/home/nik/repositories/megra.rs" :client #s(lsp--client :language-id nil :add-on? nil :new-connection (:connect #f(compiled-function (filter sentinel name environment-fn workspace) #<bytecode 0x111e6683dd226645>) :test\? #f(compiled-function () #<bytecode -0x1dd7b1929c6041bb>)) :ignore-regexps nil :ignore-messages nil :notification-handlers #<hash-table equal 1/65 0x15851f4c3521> :request-handlers #<hash-table equal 0/65 0x15851f4c3adb> :response-handlers #<hash-table eql 0/65 0x15851f4c3dc7> :prefix-function nil :uri-handlers #<hash-table equal 0/65 0x15851f4c40d5> :action-handlers #<hash-table equal 4/65 0x15851f4c3805> :major-modes nil :activation-fn #f(compiled-function (file-name mode) #<bytecode -0x6bb9f9640c09281>) :priority 1 :server-id rust-analyzer :multi-root nil :initialization-options lsp-rust-analyzer--make-init-options :semantic-tokens-faces-overrides (:discard-default-modifiers t :modifiers (... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ...)) :custom-capabilities ((experimental ...)) :library-folders-fn #f(compiled-function (workspace) #<bytecode -0x47e8177e0b048f8>) :before-file-open-fn nil :initialized-fn nil :remote? nil :completion-in-comments? nil :path->uri-fn nil :uri->path-fn nil :environment-fn nil :after-open-fn nil :async-request-handlers #<hash-table equal 0/65 0x15851f4c43ab> :download-server-fn #f(compiled-function (client callback error-callback update\?) #<bytecode 0x1730d1c2ae8ef963>) :download-in-progress? nil :buffers nil :synchronize-sections nil) :host-root nil :proc #<process rust-analyzer> :cmd-proc #<process rust-analyzer> :buffers (#<buffer osc_client.rs>) :semantic-tokens-faces nil :semantic-tokens-modifier-faces nil :extra-client-capabilities nil :status initialized :metadata #<hash-table equal 0/65 0x15851f52ae5f> :watches #<hash-table equal 0/65 0x15851f53ef47> :workspace-folders nil :last-id 0 :status-string nil :shutdown-action nil :diagnostics #<hash-table equal 6/65 0x15851f4ec38d> :work-done-tokens #<hash-table equal 0/65 0x15851f521f8f>) . [#<hash-table equal 3/3 0x15851f72f8c1>])))
  #f(compiled-function (result) #<bytecode 0x420e5d7d0def752>)([#<hash-table equal 3/3 0x15851f72f8c1>])
  #f(compiled-function (result) #<bytecode -0x7e364643ace1468>)([#<hash-table equal 3/3 0x15851f72f8c1>])
  lsp--parser-on-message(#<hash-table equal 3/3 0x15851f72f54b> #s(lsp--workspace :ewoc #s(ewoc :buffer #<buffer *lsp-log: rust-analyzer:48051*> :pretty-printer lsp--log-entry-pp :header [[... #4 DL-LIST #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] [#4 ... ... #<marker at 1 in *lsp-log: rust-analyzer:48051*>] "" #<marker at 1 in *lsp-log: rust-analyzer:48051*>] :footer [[... #4 ... #<marker at 27436 in *lsp-log: rust-analyzer:48051*>] [#4 ... DL-LIST #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] "" #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] :dll [[... #4 "" #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] [#4 ... "" #<marker at 1 in *lsp-log: rust-analyzer:48051*>] DL-LIST #<marker at 27993 in *lsp-log: rust-analyzer:48051*>] :last-node nil :hf-pp insert) :server-capabilities #<hash-table equal 26/26 0x15851f554a9b> :registered-server-capabilities (#s(lsp--registered-capability :id "workspace/didChangeWatchedFiles" :method "workspace/didChangeWatchedFiles" :options #<hash-table equal 1/1 0x15851f811829>) #s(lsp--registered-capability :id "workspace/didChangeWatchedFiles" :method "workspace/didChangeWatchedFiles" :options #<hash-table equal 1/1 0x15851f6d113b>)) :root "/home/nik/repositories/megra.rs" :client #s(lsp--client :language-id nil :add-on? nil :new-connection (:connect #f(compiled-function (filter sentinel name environment-fn workspace) #<bytecode 0x111e6683dd226645>) :test\? #f(compiled-function () #<bytecode -0x1dd7b1929c6041bb>)) :ignore-regexps nil :ignore-messages nil :notification-handlers #<hash-table equal 1/65 0x15851f4c3521> :request-handlers #<hash-table equal 0/65 0x15851f4c3adb> :response-handlers #<hash-table eql 0/65 0x15851f4c3dc7> :prefix-function nil :uri-handlers #<hash-table equal 0/65 0x15851f4c40d5> :action-handlers #<hash-table equal 4/65 0x15851f4c3805> :major-modes nil :activation-fn #f(compiled-function (file-name mode) #<bytecode -0x6bb9f9640c09281>) :priority 1 :server-id rust-analyzer :multi-root nil :initialization-options lsp-rust-analyzer--make-init-options :semantic-tokens-faces-overrides (:discard-default-modifiers t :modifiers (... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ...)) :custom-capabilities ((experimental ...)) :library-folders-fn #f(compiled-function (workspace) #<bytecode -0x47e8177e0b048f8>) :before-file-open-fn nil :initialized-fn nil :remote? nil :completion-in-comments? nil :path->uri-fn nil :uri->path-fn nil :environment-fn nil :after-open-fn nil :async-request-handlers #<hash-table equal 0/65 0x15851f4c43ab> :download-server-fn #f(compiled-function (client callback error-callback update\?) #<bytecode 0x1730d1c2ae8ef963>) :download-in-progress? nil :buffers nil ...) :host-root nil :proc #<process rust-analyzer> :cmd-proc #<process rust-analyzer> :buffers (#<buffer osc_client.rs>) :semantic-tokens-faces nil :semantic-tokens-modifier-faces nil :extra-client-capabilities nil :status initialized :metadata #<hash-table equal 0/65 0x15851f52ae5f> :watches #<hash-table equal 0/65 0x15851f53ef47> :workspace-folders nil :last-id 0 :status-string nil :shutdown-action nil :diagnostics #<hash-table equal 6/65 0x15851f4ec38d> :work-done-tokens #<hash-table equal 0/65 0x15851f521f8f>))
  #f(compiled-function (msg) #<bytecode 0x1e65bbe9f58ef4ae>)(#<hash-table equal 3/3 0x15851f72f54b>)
  mapc(#f(compiled-function (msg) #<bytecode 0x1e65bbe9f58ef4ae>) (#<hash-table equal 3/3 0x15851f72f54b>))
  #f(compiled-function (proc input) #<bytecode -0x1dbb10cccfb33145>)(#<process rust-analyzer> "Content-Length: 273\15\n\15\n{\"jsonrpc\":\"2.0\",\"id\":74,\"r...")
```

**Comment by the-drunk-coder** [comment]
Hmm ok, using the `lsp-use-plists` flag as described here: https://emacs-lsp.github.io/lsp-mode/page/performance/
seems to solve the issue.

**Comment by yyoncho** [comment]
That happens if you have this flag inconsistent state.

**Comment by sellout** [comment]
I ran into this same callstack, except for deno instead of rust-analyzer.

I don’t _think_ the flag is inconsistent in my case – it should be unset in the environment and it’s nil in emacs.

If I intentionally set `lsp-use-plists` without running emacs under `LSP_USE_PLISTS`, creating an inconsistent state, then I get the same error in many more locations.

Digging in a bit (and I certainly don’t grok all the plist/ht magic yet), the `range` hash-table contains plists for the positions. `range` is created via `lsp-lens--window-range`, which calls `lsp--point-to-position`, which calls `lsp--cur-position`.

`lsp--cur-position` explicitly builds a plist
```elisp
(list :line (lsp--cur-line)
      :character (- (point) (line-beginning-position)))
```
instead of using `lsp-make-position`. I see a couple other places do the same thing (e.g., `lsp--range`), so maybe this isn’t the bug.

I made a quick change, `s/list/lsp-make-position/`, which causes `lsp--position-compare` to be called with two hash tables, rather than a plist and a hash table, which gets me a bit farther. However, the range’s `:end` position has `nil` as the `:line` and `:character` values (the `:start` position is correct). So it’s still broken.

I’m stopping here because I don’t know if I’m headed the right direction here.

Disabling `lsp-lens-mode` makes it all go away, so I’m just doing that for now.
